firmware/thruna/BMS/CMakeLists.txt

```diff
@@ -50,6 +50,32 @@ list(REMOVE_ITEM GOOGLETEST_TEST_SRCS "${CMAKE_CURRENT_SOURCE_DIR}/Test/Src/main
 set(GOOGLETEST_TEST_INCLUDE_DIRS
     "${CMAKE_CURRENT_SOURCE_DIR}/Test/Inc")
 
+# INCLUDES and SOURCES from shared
+set(SHARED_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../shared")
+file(GLOB_RECURSE SHARED_APP_SRCS "${SHARED_DIR}/src/app/*.c")
+file(GLOB_RECURSE SHARED_IO_SRCS "${SHARED_DIR}/src/io/*.c")
+file(GLOB_RECURSE SHARED_HW_SRCS "${SHARED_DIR}/src/hw/*.c")
+
+set(SHARED_APP_INCLUDE_DIRS "${SHARED_DIR}/src/app")
+set(SHARED_IO_INCLUDE_DIRS
+    "${SHARED_DIR}/src/io"
+    "${SHARED_DIR}/src/hw"
+)
```

Comment:
can we set this in `firmware/CMakeLists.txt` rather than pass it all the way down the function? This saves a lot more code between all the files in `firemware/thruna/<BOARD>/CmakeLists.txt`

Reply:
Because from what I can tell the variables are the same throughout.

Reply:
By the way variables defined at higher levels of the folder structure carry over to lower levels.

Reply:
The primary I made this change is this.
I am currently writing an SD_card driver that is based on the HAL SD_Card driver for f4dev. Gus told me to put this in the shared folder and I think it makes sense because other boards might use this code in the future.  But, if I set it up like this, all other boards can't compile because they do not have the HAL driver for the SD card. 

Also, I am setting up a new board f4dev and I only want it to have HAL SD_Card driver on the project for now. But, if I want to build the binary, I have to add a bunch of other drivers and middleware to solve the dependency issue just to compile the shared code.  

So I want each individual board to have the ability to select the shared code it needs; so that other existing boards can include everything in shared code but exclude the SD_Card driver, and the new f4dev board can only include the SD_card driver and exclude anything else.

In order to do that, I think it has to be done in the CmakeLists.txt for each board.

Reply:
Can you make it so that `SHARED_APP_SRCS`, `SHARED_IO_SRCS`, `SHARED_HW_SRCS` in `firmare/CMakeLists.txt` and set `ALL_SHARED_SRCS` and `ALL_SHARED_INCLUDE_DIRS` in each CMakeLists.txt?

Reply:
So you pick everything you want from each board's CMakeLists.txt but you keep the helper variables in the higher context.

Reply:
The changes are pushed.